src/main/java/racingcar/domain/car/Cars.java

```diff
@@ -0,0 +1,68 @@
+package racingcar.domain.car;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import racingcar.util.RandomGenerator;
+import racingcar.view.input.error.InputError;
+import racingcar.view.input.error.InputIllegalArgumentException;
+
+public class Cars {
+
+    private static final String CARS_ARE_EMPTY = "차가 존재하지 않습니다.";
+    public static final String DISTANT_MARK = "-";
+
+    private final List<Car> cars = new ArrayList<>();
+
+    public Cars(List<String> carNames) {
+        validateCanRace(carNames);
+
+        for (String name : carNames) {
+            cars.add(new Car(name));
+        }
+    }
```

Comment:
다른 부분 처럼 이 부분도 스트림을 사용하면 어떨까요??

```suggestion
    public Cars(List<String> carNames) {
        validateCanRace(carNames);
        this.cars = carNames.stream()
                            .map(name -> new Car(name)))
                            .collect(Collectors.toList());
    }
```

Reply:
우선 이와 같이 순차적으로 `add`만하는 경우
성능상의 차이는 없고 가독성의 차이가 있다고 생각했는데요!
왜인지 해당 부분의 경우 `stream`보다 `for`문이 가독성이 좋다고 판단한 것 같아요!
그치만 이부분은 시선의 차이가 있는 것 같아
@Hyeon0208  님은 어떻게 생각하시는지 궁금합니다!! 🥰🥰

Reply:
@jinchiim 저도 같은 부분에서 고민했어요. 그래서 제가 알게 된 지식을 공유하고 싶습니다.

`carNames` 매개 변수는 List입니다. List는 `.forEach()` 를 통해 반복문을 돌릴 수 있어요. 

이를 활용해 Car객체를 할당한다면, 조금 더 가독성을 높일 수 있지 않을까 합니다. 😄

Reply:
참고로 stream에서 .collect(Collector.toList()) 대신 .toList() 를 사용할 수 있습니다~ 
하지만 후자의 경우에는 List 자체에 대한 변경이 불가능합니다

Reply:
두 분다 의견 정말 감사합니다!!
맞습니다! `forEach()`를 사용해서 반복문 돌리기가 가능하더라고요!
다만 해당 부분에서도 성능상의 차이가 발생하지 않는다고 생각해서 위와 같이 작성하게 되었어요!
@Hugh-KR 님께서는 `.forEach()`가 가독성이 더 좋다고 생각하신 것 같은데 맞을까요?! 아니면 혹시 제가 모르는 성능상의 차이가 존재하는지 궁금합니다! 🥰

> 참고로 stream에서 .collect(Collector.toList()) 대신 .toList() 를 사용할 수 있습니다~
하지만 후자의 경우에는 List 자체에 대한 변경이 불가능합니다

`toList()`가 있다는 것을 이번에 `Seol-JY`님과 `hoonyworld`님을 통해서 알게 되었는데요!
만약 위와 같은 상황이라면 불변이어야 하기 때문에 `.toList()`를 사용하는게 더 어울렸겠다는 생각이 드네요!
두분 다 좋은 지식 좋은 의견 감사합니다!
3주차때는 그부분을 더 신경써보겠습니다! 🥰